### ISSUE TITLE
Search Autocomplete Instead of Instant Search

### ISSUE THREAD

**Issue by dajinchu** [ryanhugh/searchneu#79]
I think autocomplete might be a better user experience than instant search results.

**Instant Search:**
 - Saves time if you know exactly what you're looking for (spell the class right or exact course code).
 - Causes the screen to flash around which is distracting and at times confusing. 

**Autocomplete:**
 - Better for exploratory search because it shows related terms to your current search, while still allowing you to deliberately choose what to search for. In other words, we have a "fuzzy" search within the suggestions box, but a more precise search once you hit enter, so you are more likely to be satisfied by the search results.
 - Scoped/filtered search can be directly suggested. ex: if you type in "Matthias" it can offer to scope your search to just employees, or if you type "advanced writing online" it can search for "advanced writing" with an "online" filter.
 - You have to hit enter.

Some articles:
Google dropped Instant Search: https://gizmodo.com/rip-google-instant-search-you-were-never-necessary-1797281069
Good UX info on autocomplete: https://baymard.com/blog/autocomplete-design

Let me know what you think. I can work on a bit of prototype to see how it feels.

**Comment by ryanhugh** [comment]
I agree this would be super cool. We can use the same search backend (and endpoint) too, instead of creating a different search pipeline for this. The current search is super fast and I'm not worried about speed at all.  

However, I wonder if we could spend our time building out other features first? Maybe it would be better to do #58 which would let us add filtering tot search (#1) first? Maybe we could add some social features? But if you want to

**Comment by dajinchu** [comment]
Yeah elasticsearch is the prereq for this for sure
